### PR TITLE
Add python hint to bigdl.orca.data 

### PR DESCRIPTION
**Pull request by leonardozcm**
## Description

This is a preview of a small part of the entire project before adding python type hint, including templates, tools and modification processes

### 1. Why the change?

In order to standardize the development process and improve the user experience

### 2. User API changes

For user: no
For developer: adapt and run `add_type_hint.sh test_dir` to generate diff database of add hints, more details refer to [Monkeytype](https://monkeytype.readthedocs.io/en/latest/)

### 3. Summary of the change 

1. Add python hint to bigdl.orca.data 
2. Provide a way to help add hints, which is:

-  pip install mypy and monkeytype
-  Auto-annotate our code by leveraging our test files,  you can use add_type_hint.sh to do this for example 

`add_type_hint.sh test_dir data` will test pytest under bigdl/orca/test/data/ and generate an `orca_hint.sqlite3` including diffs monkeytype collected. You can also [do it manually](https://monkeytype.readthedocs.io/en/latest/#example)
- Apply changes(monkeytype stub, for there are issues with monkeytype apply )

### 4. How to test?
- [ ] N/A
- [x] Unit test
- [ ] Application test
- [ ] Document test
- [ ] ...

### 5. New dependencies

- [x] New Python dependencies
       - mypy(test only)
       - monkeytype(only for developers)
       - ...

### CODE REVIEW

**Comment by leonardozcm** [comment]
By running mypy, you will see 
```
(py37) arda@intern03:~/Project/BigDL/python/orca$ mypy --config-file test/mypy.ini src/bigdl/orca/data/utils.py src/bigdl/orca/data/ray_xshards.py
Success: no issues found in 2 source files
```
if type checking passed.

File: python/orca/src/bigdl/orca/data/image/parquet_dataset.py [hunk 1]
```diff
@@ -32,11 +32,23 @@
...
+    from bigdl.orca.data.shard import SparkXShards
+    from pyspark.rdd import PipelinedRDD
+    from bigdl.orca.data.tf.data import TensorSliceDataset, Dataset
 
 
 class ParquetDataset:
     @staticmethod
-    def write(path, generator, schema, block_size=1000, write_mode="overwrite", **kwargs):
+    def write(path: str,
+              generator: List[Dict[str, Union[str, int, ndarray]]],
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/data/image/parquet_dataset.py:47]: int and float

File: python/orca/src/bigdl/orca/data/image/parquet_dataset.py [hunk 2]
```diff
@@ -118,13 +130,13 @@ def merge_records(schema, iter):
...
     @staticmethod
-    def read_as_tf(path):
+    def read_as_tf(path: str) -> "TensorSliceDataset":
         """
         return a orca.data.tf.data.Dataset
         :param path:
         :return:
         """
-        from bigdl.orca.data.tf.data import Dataset
+        from bigdl.orca.data.tf.data import TensorSliceDataset, Dataset
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/data/image/parquet_dataset.py:139]: why

File: python/orca/src/bigdl/orca/data/image/parquet_dataset.py [hunk 3]
```diff
@@ -201,12 +213,12 @@ def __call__(self):
...
 
 
-def _read32(bytestream):
+def _read32(bytestream: io.BufferedReader) -> uint32:
     dt = np.dtype(np.uint32).newbyteorder('>')
     return np.frombuffer(bytestream.read(4), dtype=dt)[0]
 
 
-def _extract_mnist_images(image_filepath):
+def _extract_mnist_images(image_filepath: str):
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/data/image/parquet_dataset.py:221]: -> ndarray

**Comment by leonardozcm** [comment]
```
def convert_imagenet_to_tf_records(
        raw_data_dir: str,
        output_dir: str) -> Tuple[List[str], List[str]]:
```
should return None

https://github.com/intel-analytics/BigDL/pull/5299/files#diff-f7838e0bb04f8222e3c77ca2e67f75e96d88a986cf86dddbe41ebd22b2a713faR58

**Comment by leonardozcm** [comment]
Add type hint to orca.data.transformer as well

File: python/orca/src/bigdl/orca/data/tf/tf1_data.py [hunk 4]
```diff
@@ -105,7 +110,7 @@ def _get_training_data(self):
                              self.inter_threads, self.intra_threads)
         return FeatureSet(jvalue=jvalue)
 
-    def _get_validation_data(self):
+    def _get_validation_data(self) -> None:
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/data/tf/tf1_data.py:113]: Optional[FeatureSet]

**Comment by leonardozcm** [comment]
orca.data.tf.tf2_data

File: python/orca/src/bigdl/orca/data/ray_xshards.py [hunk 5]
```diff
@@ -234,7 +255,10 @@ def transform_shards_with_actors(self, actors, func):
 
         return RayXShards.from_partition_refs(actor_ip2part_id, new_part_id_refs, self.rdd)
 
-    def reduce_partitions_for_actors(self, actors, reduce_partitions_func, return_refs=False):
+    # TODO: ADD UT
+    def reduce_partitions_for_actors(self, actors: List["ActorHandle"],
+                                     reduce_partitions_func: Callable,
+                                     return_refs: bool=False):
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/data/ray_xshards.py:261]: -> List[Any]

File: python/orca/src/bigdl/orca/data/ray_xshards.py [hunk 6]
```diff
@@ -262,8 +286,12 @@ def reduce_partitions_for_actors(self, actors, reduce_partitions_func, return_re
...
         return results
 
-    def zip_reduce_shards_with_actors(self, xshards, actors, reduce_partitions_func,
-                                      return_refs=False):
+    # TODO: ADD UT
+    def zip_reduce_shards_with_actors(self,
+                                      xshards: "RayXShards",
+                                      actors: List["ActorHandle"],
+                                      reduce_partitions_func,
+                                      return_refs: bool=False):
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/data/ray_xshards.py:294]: reduce_partitions_func:Callable
->List[Any]

File: python/orca/src/bigdl/orca/data/shard.py [hunk 7]
```diff
@@ -422,16 +451,16 @@ def deduplicates(self):
             invalidInputError(False,
                               "Currently only support dedup() on XShards of Pandas DataFrame")
 
-    def assembleFeatureLabelCols(self, featureCols, labelCols):
+    def assembleFeatureLabelCols(self,
+                                 featureCols: List[Union[str, "Column"]],
+                                 labelCols: List[Union[str, "Column"]]):
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/data/shard.py:456]: -> "SparkXShards"

File: python/orca/src/bigdl/orca/data/tf/data.py [hunk 8]
```diff
@@ -116,26 +125,26 @@ def get_xshards(self):
             return new_shards
 
     @staticmethod
-    def from_tensor_slices(xshards):
+    def from_tensor_slices(xshards: "SparkXShards") -> "TensorSliceDataset":
         return TensorSliceDataset(xshards)
 
     @staticmethod
-    def from_feature_table(tbl):
+    def from_feature_table(tbl: "FeatureTable"):
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/data/tf/data.py:132]: -> "TensorSliceDataset":

File: python/orca/src/bigdl/orca/data/tf/tf1_data.py [hunk 9]
```diff
@@ -20,12 +20,17 @@
...
 
 class TF1Dataset(TFDataset):
 
-    def __init__(self, dataset, batch_size,
-                 batch_per_thread,
-                 validation_dataset=None, intra_threads=None, inter_threads=None):
+    def __init__(self, dataset: "MapDataset", batch_size: int,
+                 batch_per_thread: int,
+                 validation_dataset: "MapDataset"=None,
+                 intra_threads: int=None, inter_threads: None=None) -> None:
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/data/tf/tf1_data.py:33]: inter_threads: int=None

File: python/orca/src/bigdl/orca/data/transformer.py [hunk 10]
```diff
@@ -265,12 +292,12 @@ def __init__(self, df, col_name):
...
         self.df = df
 
-    def broadcast(self):
+    def broadcast(self) -> None:
         from pyspark.sql.functions import broadcast
         self.df = broadcast(self.df)
 
     @classmethod
-    def from_dict(cls, indices, col_name):
+    def from_dict(cls, indices, col_name) -> "StringIndex":
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/data/transformer.py:300]: indices: Dict[str, int]
col_name: str

File: python/orca/src/bigdl/orca/data/image/parquet_dataset.py [hunk 11]
```diff
@@ -419,7 +450,12 @@ def worker_init_fn(w_id):
                                        batch_size=batch_size, worker_init_fn=worker_init_fn)
 
 
-def read_parquet(format, path, transforms=None, config=None, batch_size=1, *args, **kwargs):
+def read_parquet(format: str,
+                 path: str,
+                 transforms: Callable=None,
```
Comment (yangw1234) [inline python/orca/src/bigdl/orca/data/image/parquet_dataset.py:455]: `Optional[Callable]` ?

File: python/orca/src/bigdl/orca/data/image/voc_dataset.py [hunk 12]
```diff
@@ -38,10 +42,10 @@ class VOCDatasets:
     difficult: bool, False ignore voc xml difficult value.
     """
 
-    def __init__(self, root="VOCdevkit",
-                 splits_names=[(2007, "trainval")],
-                 classes=None,
-                 difficult=False) -> None:
+    def __init__(self, root: str="VOCdevkit",
+                 splits_names: List[Tuple[int, str]]=[(2007, "trainval")],
+                 classes: None=None,
```
Comment (yangw1234) [inline python/orca/src/bigdl/orca/data/image/voc_dataset.py:47]: `Optional[List[str]]`?

**Comment by yangw1234** [comment]
Do we have type checking tests for orca?

**Comment by leonardozcm** [comment]
> Do we have type checking tests for orca?

I had enable mypy for type checking on orca, but actually it seems not that useful on orca, for there are a lot of types in a tangle, it would be a great work to fix our codestyle. So at this stage, we only apply typehints on user APIs manually.

**Comment by yangw1234** [comment]
> > Do we have type checking tests for orca?
> 
> I had enable mypy for type checking on orca, but actually it seems not that useful on orca, for there are a lot of types in a tangle, it would be a great work to fix our codestyle. So at this stage, we only apply typehints on user APIs manually.

Is it possible to first enable the type checking in `orca.data`. Otherwise it is hard to verify the changes are correct.

**Comment by leonardozcm** [comment]
> Is it possible to first enable the type checking in `orca.data`. Otherwise it is hard to verify the changes are correct.

Sure, I will try this in `orca.data`.

File: python/orca/src/bigdl/orca/data/shard.py [hunk 13]
```diff
@@ -18,9 +18,11 @@
...
-from bigdl.dllib.utils.common import *
+from bigdl.dllib.utils.common import (get_node_and_core_number,
+                                      get_spark_sql_context,
+                                      get_spark_context)
 from bigdl.dllib.utils import nest
-from bigdl.dllib.utils.log4Error import *
+from bigdl.dllib.utils.log4Error import invalidInputError
 
 import numpy as np
 from pandas.core.frame import DataFrame as PandasDataFrame
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/shard.py:28]: remove this?

File: python/orca/src/bigdl/orca/data/file.py [hunk 14]
```diff
@@ -205,7 +211,7 @@ def makedirs(path):
         return os.makedirs(path)
 
 
-def write_text(path, text):
+def write_text(path: str, text: str):
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/file.py:214]: where's the return value?

File: python/orca/src/bigdl/orca/data/file.py [hunk 15]
```diff
@@ -385,7 +391,7 @@ def put_local_file_to_remote(local_path, remote_path, filemode=None):
                 process = subprocess.Popen(chmod_cmd, shell=True)
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/file.py:391]: Why the functions between Line 217 and 391 are not typed?

File: python/orca/src/bigdl/orca/data/file.py [hunk 16]
```diff
@@ -403,7 +409,7 @@ def put_local_file_to_remote(local_path, remote_path, filemode=None):
             with open(local_path, "rb") as f:
                 s3_client.upload_fileobj(f, Bucket=bucket, Key=prefix)
         except Exception as e:
-            logger.error("Cannot upload file {} to {}: error: "
+            logger.error("Cannot upload file {} to {}: error: {}"
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/file.py:412]: why these functions are not typed?

File: python/orca/src/bigdl/orca/data/image/imagenet_dataset.py [hunk 17]
```diff
@@ -136,7 +141,7 @@ def cmyk_to_rgb(self, image_data: bytes):
         return self._sess.run(self._cmyk_to_rgb,
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/image/imagenet_dataset.py:141]: why this is not typed?

File: python/orca/src/bigdl/orca/data/file.py [hunk 18]
```diff
@@ -21,12 +21,18 @@
 import functools
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/file.py:21]: The function in this file better all be typed. Maybe used by users.

File: python/orca/src/bigdl/orca/data/image/voc_dataset.py [hunk 19]
```diff
@@ -72,24 +76,25 @@ def _load_items(self, splits_names):
...
-            img = self._im_cache
+            img = self._im_cache[img_path]
         else:
-            img = self._read_image(img_path)
+            img = self._read_image(img_path)  # type: ignore
 
         return img, self._im_anno[idx]
 
-    def _load_label(self, idx):
+    @no_type_check
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/image/voc_dataset.py:96]: what does this mean? is it different from type: ignore?

File: python/orca/src/bigdl/orca/data/pandas/preprocessing.py [hunk 20]
```diff
@@ -19,10 +19,16 @@
...
 from bigdl.orca.data import SparkXShards
 from bigdl.orca.data.utils import *
-from bigdl.dllib.utils.log4Error import *
+from bigdl.dllib.utils.log4Error import invalidInputError
+from typing import TYPE_CHECKING, List, Optional
 
+if TYPE_CHECKING:
+    from bigdl.orca.data.shard import SparkXShards
+    from pyspark.sql.types import StructType
+    import os
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/pandas/preprocessing.py:28]: why need import os?

File: python/orca/src/bigdl/orca/data/shard.py [hunk 21]
```diff
@@ -18,25 +18,48 @@
...
+
+if TYPE_CHECKING:
+    from numpy import ndarray
+    from pandas.core.frame import DataFrame as PandasDataFrame
+    from pandas.core.indexes.base import Index
+    from pandas.core.series import Series
+    from pyspark.rdd import (
+        PipelinedRDD,
+        RDD
+    )
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/shard.py:51]: put this into one line?

File: python/orca/src/bigdl/orca/data/shard.py [hunk 22]
```diff
@@ -18,25 +18,48 @@
...
+
 
 class XShards(object):
     """
 
     A collection of data which can be pre-processed in parallel.
     """
-
-    def transform_shard(self, func, *args):
+    def transform_shard(self, func: Callable, *args):
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/shard.py:62]: add return value

File: python/orca/src/bigdl/orca/data/shard.py [hunk 23]
```diff
@@ -143,8 +172,10 @@ class SparkXShards(XShards):
 
     A collection of data which can be pre-processed in parallel on Spark
     """
-
-    def __init__(self, rdd, transient=False, class_name=None):
+    def __init__(self,
+                 rdd: Union["PipelinedRDD", "RDD"],
+                 transient: bool=False,
+                 class_name=None) -> None:
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/shard.py:178]: missing type, should be str?

File: python/orca/src/bigdl/orca/data/shard.py (hunk 22 above)
Comment (hkvision) [reply python/orca/src/bigdl/orca/data/shard.py:62]: or can ignore in the XShards class as no one will directly use it?

File: python/orca/src/bigdl/orca/data/shard.py [hunk 24]
```diff
@@ -797,7 +837,7 @@ def utility_func(x, func, *args, **kwargs):
         result_rdd = self.rdd.map(lambda x: utility_func(x, func, *args, **kwargs))
         return result_rdd
 
-    def get_schema(self):
+    def get_schema(self) -> Optional[Dict[str, Union["Index", "Series"]]]:
         if 'schema' in self.type:
             return self.type['schema']
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/shard.py:843]: missing type for `merge`

File: python/orca/src/bigdl/orca/data/shard.py [hunk 25]
```diff
@@ -905,7 +945,7 @@ def f(iterator):
 
 
 class SharedValue(object):
-    def __init__(self, data):
+    def __init__(self, data: int) -> None:
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/shard.py:948]: this is wrong? not necessarily int?

File: python/orca/src/bigdl/orca/data/tf/data.py [hunk 26]
```diff
@@ -30,12 +36,15 @@ class Dataset(object):
...
+                 create_dataset_fn: Callable,
+                 xshards_transform_fn: Optional[Callable]=None) -> None:
         self.xshards = xshards
         self.create_dataset_fn = create_dataset_fn
         self.xshards_transform_fn = xshards_transform_fn
 
-    def as_graph_rdd(self, batch_per_shard, drop_remainder=True):
+    def as_graph_rdd(self, batch_per_shard: int, drop_remainder: bool=True) -> "RDD[Any]":
 
         create_dataset_fn = self.create_dataset_fn
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/tf/data.py:50]: as_tf_dataset_rdd should be typed.

File: python/orca/src/bigdl/orca/data/tf/data.py [hunk 27]
```diff
@@ -116,26 +125,26 @@ def get_xshards(self):
             return new_shards
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/tf/data.py:125]: get_xshards should be typed

File: python/orca/src/bigdl/orca/data/transformer.py [hunk 28]
```diff
@@ -292,13 +321,13 @@ def from_dict(cls, indices, col_name):
...
-        indices = map(lambda x: {col_name: x[0], 'id': x[1]}, indices.items())
+        new_indices = map(lambda x: {col_name: x[0], 'id': x[1]}, indices.items())
         schema = StructType([StructField(col_name, StringType(), False),
                              StructField("id", IntegerType(), False)])
-        df = spark.createDataFrame((Row(**x) for x in indices), schema=schema)
+        df = spark.createDataFrame((Row(**x) for x in new_indices), schema=schema)
         return cls(df, col_name)
 
-    def to_dict(self):
+    def to_dict(self) -> Dict[str, "StringIndex"]:
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/transformer.py:330]: should be dict(str, int)?

File: python/orca/src/bigdl/orca/data/transformer.py [hunk 29]
```diff
@@ -316,43 +345,51 @@ def to_dict(self):
...
         return data_shards
 
-    def transform(self, shard):
+    def transform(self, shard: "SparkXShards") -> "SparkXShards":
         invalidInputError(self.scalerModel, "Please call fit_transform first")
         df = shard.to_spark_df()
-        scaledData = self.scalerModel.transform(df)
+        scaledData = self.scalerModel.transform(df)  # type: ignore
         data_shards = spark_df_to_pd_sparkxshards(scaledData)
         return data_shards
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/transformer.py:394]: StandardScalar is not typed.

File: python/orca/src/bigdl/orca/data/utils.py [hunk 30]
```diff
@@ -403,8 +425,8 @@ def spark_df_to_rdd_pd(df, squeeze=False, index_col=None,
         return pd_rdd
 
 
-def spark_df_to_pd_sparkxshards(df, squeeze=False, index_col=None,
-                                dtype=None, index_map=None):
+def spark_df_to_pd_sparkxshards(df: "DataFrame", squeeze: bool=False, index_col: None=None,
+                                dtype: None=None, index_map: None=None) -> "SparkXShards":
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/utils.py:429]: why these types are None?

File: python/orca/src/bigdl/orca/data/utils.py [hunk 31]
```diff
@@ -458,7 +480,12 @@ def f(iter):
     return f
 
 
-def to_pandas(df, squeeze=False, index_col=None, dtype=None, index_map=None, batch_size=None):
+def to_pandas(columns: List[str], squeeze: bool=False, index_col: Optional[str]=None,
+              dtype: Optional[Union[str, Dict[str, Type[float32]],
+                                    Dict[int, Union[Type[float32], Type[int32]]],
+                                    Dict[str, str]]]=None,
+              index_map: Optional[Dict[int, str]]=None,
+              batch_size: None=None) -> Callable:
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/utils.py:488]: why is None?

Comment (hkvision) [reply python/orca/src/bigdl/orca/data/utils.py:488]: return is RDD not Callable

File: python/orca/src/bigdl/orca/data/utils.py [hunk 32]
```diff
@@ -376,8 +396,10 @@ def get_size(x):
                           " or a list of ndarrays, please check your input")
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/data/utils.py:396]: Add type for process_spark_xshards?

**Comment by hkvision** [comment]
https://github.com/intel-analytics/BigDL/blob/main/python/orca/src/bigdl/orca/data/elastic_search.py is missing, is it needed?

**Comment by hkvision** [comment]
Do we have any mechanism to make sure if future functions are added in orca/data they are always typed?

File: python/orca/src/bigdl/orca/data/file.py (hunk 15 above)
Comment (leonardozcm) [reply python/orca/src/bigdl/orca/data/file.py:391]: > Why the functions between Line 217 and 391 are not typed?

Not sure if these are needed, add now

File: python/orca/src/bigdl/orca/data/file.py (hunk 18 above)
Comment (leonardozcm) [reply python/orca/src/bigdl/orca/data/file.py:21]: done then

File: python/orca/src/bigdl/orca/data/image/voc_dataset.py (hunk 19 above)
Comment (leonardozcm) [reply python/orca/src/bigdl/orca/data/image/voc_dataset.py:96]: > what does this mean? is it different from type: ignore?
This annotation just ignores type checking of this method, while type:ignore only works for one line.

File: python/orca/src/bigdl/orca/data/pandas/preprocessing.py (hunk 20 above)
Comment (leonardozcm) [reply python/orca/src/bigdl/orca/data/pandas/preprocessing.py:28]: > why need import os?

Actually in preprocessing.py, `import os` is missing. So it is needed and out of this block.

File: python/orca/src/bigdl/orca/data/shard.py (hunk 22 above)
Comment (leonardozcm) [reply python/orca/src/bigdl/orca/data/shard.py:62]: > or can ignore in the XShards class as no one will directly use it?

It has been ignored here

File: python/orca/src/bigdl/orca/data/shard.py (hunk 25 above)
Comment (leonardozcm) [reply python/orca/src/bigdl/orca/data/shard.py:948]: > this is wrong? not necessarily int?

yes, how about my ignoring this?

File: python/orca/src/bigdl/orca/data/tf/data.py [hunk 33]
```diff
@@ -30,12 +36,15 @@ class Dataset(object):
...
+    def __init__(self,
+                 xshards: "SparkXShards",
+                 create_dataset_fn: Callable,
+                 xshards_transform_fn: Optional[Callable]=None) -> None:
         self.xshards = xshards
         self.create_dataset_fn = create_dataset_fn
         self.xshards_transform_fn = xshards_transform_fn
 
-    def as_graph_rdd(self, batch_per_shard, drop_remainder=True):
+    def as_graph_rdd(self, batch_per_shard: int, drop_remainder: bool=True) -> "RDD[Any]":
```
[the remark this answers is not in the document]
Comment (leonardozcm) [reply python/orca/src/bigdl/orca/data/tf/data.py:47]: > why is it Any? is it too complicated to list the dict type?

RDD[Any] is suggested by mypy and monkeytype, i think we'd better keep this?

**Comment by leonardozcm** [comment]
> https://github.com/intel-analytics/BigDL/blob/main/python/orca/src/bigdl/orca/data/elastic_search.py is missing, is it needed?

I think it is for internal use.

**Comment by leonardozcm** [comment]
> Do we have any mechanism to make sure if future functions are added in orca/data they are always typed?

1, we can enable strict mode in mypy to force checking every type hint, but this may require a global range modification in orca(It's really hard i think, for there are a lot of dependencies we rely on).
2. do we need to type hint every method in orca? as far as i know, there are few BIG repos add type hints on every file. For now we require committers to append they modules in https://github.com/intel-analytics/BigDL/pull/5299/files#diff-f0eb7bb6f684bfeb2c36e89a05d9c8c4fc15e3c62eb1c80e44d20c4f2cd1bdabR12, we make sure **every methods annotated are correct**.

File: python/orca/src/bigdl/orca/data/tf/data.py (hunk 33 above)
[the remark this answers is not in the document]
Comment (hkvision) [reply python/orca/src/bigdl/orca/data/tf/data.py:47]: okay

**Pull request merged**